app/src/main/java/com/example/chattutorial/ChannelActivity3.kt

```diff
@@ -20,20 +21,21 @@ import kotlinx.android.synthetic.main.activity_channel_3.*
 
 class ChannelActivity3 : AppCompatActivity(R.layout.activity_channel_3) {
 
-    private val cid by lazy {
-        intent.getStringExtra(CID_KEY)
+    private val cid: String by lazy {
+        intent.getStringExtra(CID_KEY)!!
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        val viewModelProvider = ViewModelProvider(this, ChannelViewModelsFactory(cid))
 
         messageListView.setViewHolderFactory(MyMessageViewHolderFactory())
-        val messagesViewModel = MessageListViewModel(cid)
+        val messagesViewModel = viewModelProvider.get(MessageListViewModel::class.java)
             .apply {
                 bindView(messageListView, this@ChannelActivity3)
                 state.observe(
                     this@ChannelActivity3,
-                    Observer {
+                    {
```

Comment:
Since we're updating them, these `observe` calls could have a trailing lambda syntax like so (there's a bunch of these in the project):

```kotlin
state.observe(this@ChannelActivity) { ... }
```